### ISSUE TITLE
Smart Invert Colors inverts the map’s colors

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-gl-native#9859]
The [Smart Invert Colors](https://developer.apple.com/videos/play/wwdc2017/215/?time=720) feature in iOS 11 is designed to avoid inverting graphical content like images and videos. However, it does invert MGLMapView’s content, including the map, any annotation views, and the ornaments, because MGLMapView is just an ordinary UIView as far as UIKit is concerned.

We should set [`MGLMapView.accessibilityignoresinvertcolors`](https://developer.apple.com/documentation/uikit/uiview/2865843-accessibilityignoresinvertcolors) to avoid this situation, since style designers rarely expect their styles to be color-inverted. If separate “light” and “dark” styles intended for a particular application, then the application can listen for the [`UIAccessibilityInvertColorsStatusDidChange`](https://developer.apple.com/documentation/foundation/nsnotification.name/1615196-uiaccessibilityinvertcolorsstatu) notification and swap styles accordingly. (That would be worth an example.)

Should annotation views be color-inverted or should be they exempted along with the map? The developer may want annotation views to contrast well against the map, but on the other hand they may want them to match other UI in the application.

/cc @frederoni @mapbox/cartography-cats @jmkiley

**Comment by jmkiley** [comment]
We probably want to set `.accessibilityignoresinvertcolors` separately for the `MGLMapView` and its subviews. This would allow the users to customize that behavior based on their use case. The default behavior can be for the map and its subviews to ignore smart invert.

<img width="200" alt="screenshot 2017-08-28 11 37 08" src="https://user-images.githubusercontent.com/12474734/29788188-90ceacfc-8be6-11e7-9089-78c8ab0ff862.png">

I have run into an issue where the map view's colors are inverted when I screenshot it. Will look into a fix for this later.

<img width="200" src="https://user-images.githubusercontent.com/12474734/29788119-512ddafa-8be6-11e7-8d80-c8f0b903bcf3.PNG">

**Comment by jmkiley** [comment]
Addressed in https://github.com/mapbox/mapbox-gl-native/pull/9876
